paddle/operators/interp_op.cc

```diff
@@ -0,0 +1,106 @@
+limitations under the License. */
+
+#include "paddle/framework/op_registry.h"
+#include "paddle/operators/net_op.h"
+
+namespace paddle {
+namespace operators {
+
+class InterpOp : public NetOp {
+ public:
+  InterpOp(const std::string &type, const framework::VariableNameMap &inputs,
+           const framework::VariableNameMap &outputs,
+           const framework::AttributeMap &attrs)
+      : NetOp(type, inputs, outputs, attrs) {
+    PADDLE_ENFORCE_NE(Input("X"), framework::kEmptyVarName,
+                      "Input(X) of InterpOp should not be null.");
+    PADDLE_ENFORCE_NE(Input("Y"), framework::kEmptyVarName,
+                      "Input(Y) of InterpOp should not be null.");
+    PADDLE_ENFORCE_NE(Input("W"), framework::kEmptyVarName,
+                      "Input(W) of InterpOp should not be null.");
+    PADDLE_ENFORCE_NE(Output("MinusOut"), framework::kEmptyVarName,
+                      "Output(MinusOut) of InterpOp should not be null.");
+    PADDLE_ENFORCE_NE(Output("MulOut"), framework::kEmptyVarName,
+                      "Output(MulOut) of InterpOp should not be null.");
+    PADDLE_ENFORCE_NE(Output("Out"), framework::kEmptyVarName,
+                      "Output(Out) of InterpOp should not be null.");
+
+    // MinusOut = X - Y
+    auto x = Input("X");
+    auto y = Input("Y");
+    auto minus_out = Output("MinusOut");
+    AppendOp(framework::OpRegistry::CreateOp("elementwise_sub",
+                                             {{"X", {x}}, {"Y", {y}}},
+                                             {{"Out", {minus_out}}}, {}));
+
+    // MulOut = MinusOut * W = (X - Y) * W
+    auto w = Input("W");
+    auto mul_out = Output("MulOut");
+    AppendOp(framework::OpRegistry::CreateOp(
+        "elementwise_mul", {{"X", {minus_out}}, {"Y", {w}}},
+        {{"Out", {mul_out}}}, {{"axis", 0}}));
+
+    // Out = MulOut + Y = (X - Y) * W + Y = X * W + Y * (1 - W)
+    AppendOp(framework::OpRegistry::CreateOp("elementwise_add",
+                                             {{"X", {mul_out}}, {"Y", {y}}},
+                                             {{"Out", {Output("Out")}}}, {}));
+
+    CompleteAddOp(false);
+  }
+};
+
+class InterpOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  InterpOpMaker(framework::OpProto *proto, framework::OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X", "A 2-D Tensor, the first input of interp_op");
+    AddInput("Y", "A 2-D Tensor, the second input of interp_op");
+    AddInput("W", "A 1-D Tensor, the interpolated values");
+    AddOutput("MinusOut",
+              "A 2-D Tensor, the intermediate outputs, saving X - Y.")
+        .AsIntermediate();
+    AddOutput("MulOut",
+              "A 2-D Tensor, the intermediate outputs,"
+              "saving the mul mul of (X - Y) and W")
+        .AsIntermediate();
+    AddOutput("Out",
+              "A 2-D Tensor, the output of interp_op, same shape with X");
```

Comment:
Would there more detail comments about inputs and outputs? About the format, you can refer to https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/name_convention.md#opprotomaker-names

Reply:
Done.